docs/pages/guides/configuration-management/generating-models.md

```diff
@@ -0,0 +1,103 @@
+---
+layout: default
+title: Generating Models
+abstract: Generating Model 
+permalink: guides/configuration-management/generating-models
+category: configuration
+type: guides
+language: en
+---
+
+Generating Models annotation and existing custom resource definition (CRD) into Meshery. The platform supports generation from URL, csv you can generate models using either the Meshery CLI or the Meshery UI.
+
+## Generate Models Using Meshery CLI
+
+**Step 1: Install Meshery CLI**
+
+Before you can use the Meshery CLI to generate a [Model](/concepts/logical/models), you must first install it. You can install Meshery CLI by [following the instructions]({{site.baseurl}}/installation#install-mesheryctl).
+
+
+**Step 2: Generate the Model**
+
+Model can generated in 2 different format ```URL, CSV```. A template file is required that contains some required properties: Registrant, Model name, Model DisplayName, Category. The template file is only required when you use URL for generation of model from a crd. This command enable users to import their new models from CRD and existing Meshery Model
+
+<pre class="codeblock-pre">
+<div class="codeblock"><div class="clipboardjs">mesheryctl model generate -f [file/url] </div></div>
+</pre>
+
+The supported registrant for generating from URL is `github` and `artifacthub`.The URL format must be in this order.
+
+
+Registrant `Artifacthub`:
+- https://artifacthub.io/packages/search?ts_query_web={ model-name } 
+- https://istio-release.storage.googleapis.com/charts/base-1.19.0-alpha.1.tgz&sa=D&source=editors&ust=1726839249773905&usg=AOvVaw0j88gkt6FOS1LLSRCYq95X 
+
+Registrant `Github`:
+- git:://github.com/cert-manager/cert-manager/master/deploy/crds
+- https://github.com/UffizziCloud/uffizzi_controller/releases/download/uffizzi-controller-2.0.1/uffizzi-controller-2.0.1.tgz&sa=D&source=editors&ust=1726839320133140&usg=AOvVaw2AryFwXIPKFnWRjRRTApzp
+
+
+**Example :**
+
+<pre class="codeblock-pre">
+<div class="codeblock"><div class="clipboardjs">mesheryctl model generate -f "git:://github.com/cert-manager/cert-manager/master/deploy/crds" -t template.json</div></div>
+</pre>
+
+
+**Note:** A `-r` flag is present to skip registaration. If the flag is used then no new model would be registered though they would be generated and stored inside `.meshery/models` directory.
+The `template.json` can be viewed [here](#).In template.json if the field `isAnnotaion` is true then we would only consider that component as an annotation (svg icon) rather than a normal component.
+
+
+
+**Example :**
+
+
+<pre class="codeblock-pre">
+<div class="codeblock"><div class="clipboardjs">mesheryctl model generate -f "git:://github.com/cert-manager/cert-manager/master/deploy/crds" -t template.json -r</div></div>
+</pre>
+
+
```

Comment:
we can also add an example for generating through the csv file.

<pre class="codeblock-pre">
<div class="codeblock"><div class="clipboardjs">mesehryctl model generate --f /path/to/csv-drectory</div></div>
</pre>